paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -783,45 +780,12 @@ internal abstract class BaseSheetViewModel(
 
     abstract fun onUserCancel()
 
-    internal fun closeScreens() {
-        backStack.value.forEach {
-            it.onClose()
-        }
-    }
-
     private fun onUserBack() {
         clearErrorMessages()
         _mandateText.value = null
 
-        backStack.update { screens ->
-            val modifiableScreens = screens.toMutableList()
-
-            val lastScreen = modifiableScreens.removeLast()
-
-            lastScreen.onClose()
-
-            reportPaymentSheetHidden(lastScreen)
-
-            modifiableScreens.toList()
-        }
-    }
-
-    private fun resetTo(screens: List<PaymentSheetScreen>) {
-        val previousBackStack = backStack.value
-
-        backStack.value = screens
-
-        previousBackStack.forEach { oldScreen ->
-            if (oldScreen !in screens) {
-                oldScreen.onClose()
-            }
-        }
-    }
-
-    private fun PaymentSheetScreen.onClose() {
-        when (this) {
-            is Closeable -> close()
-            else -> Unit
+        navigationHandler.pop { poppedScreen ->
+            reportPaymentSheetHidden(poppedScreen)
```

Comment:
can/should we also move `reportPaymentSheetHidden` to the navigation handler? It looks like this is it's only call site

Reply:
ok that makes sense!